### PR TITLE
2.0 Release Final

### PR DESCRIPTION
**Pull request by HanlinMiao**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 [tool.poetry]
 name = "nautobot-secrets-providers"
-version = "1.4.1b1"
-description = "Nautobot Secrets Providers Plugin."
-authors = ["Network to Code, LLC <info@networktocode.com>"]
+version = "2.0.0rc1"
```
Comment (bryanculver) [inline pyproject.toml:3]:
```suggestion
version = "2.0.0"
```

Comment (bryanculver) [reply pyproject.toml:3]: This will probably need a lockfile update

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -119,26 +120,18 @@ jobs:
...
-            db-backend: "mysql"
-            nautobot-version: "1.4.0"
-          - python-version: "3.10"
-            db-backend: "mysql"
-            nautobot-version: "stable"
+        # TBD: Fix tested versions
+        # python-version: ["3.8", "3.9", "3.10", "3.11"]
+        python-version: ["3.8"]
+        nautobot-version: ["2.0.0"]
+        db-backend: ["postgresql", "mysql"]
```
Comment (bryanculver) [inline .github/workflows/ci.yml:127]:
```suggestion
        python-version: ["3.8", "3.11"]
        nautobot-version: ["2.0.0", "stable"]
        db-backend: ["postgresql", "mysql"]
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,15 @@
 # Nautobot Secrets Providers Changelog
 
+## v2.0.0 (TBD)
```
Comment (bryanculver) [inline CHANGELOG.md:3]:
```suggestion
## v2.0.0 (2023-09-29)
```

**Review by bryanculver — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (HanlinMiao) [reply pyproject.toml]: Didnt really need one it seems

**Review by bryanculver — APPROVED** [review]

**Pull request merged**